kaldb/src/test/java/com/slack/kaldb/chunkManager/IndexingChunkManagerTest.java

```diff
@@ -1084,7 +1084,7 @@ public void testMultipleByteRollOversSuccessfully()
     final long msgsPerChunk = 3L;
     final long maxBytesPerChunk = 100L;
     final ChunkRollOverStrategy chunkRollOverStrategy =
-        new ChunkRollOverStrategyImpl(maxBytesPerChunk, msgsPerChunk);
+        new SizeOrMessageBasedRolloverStrategy(maxBytesPerChunk, msgsPerChunk);
```

Comment:
Also, can you please add a new unit test for the current failure condition. I think creating a chunk with a lot of messages and not waiting for rollover synchronously  would recreate the failure.